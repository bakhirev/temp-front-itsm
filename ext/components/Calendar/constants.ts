export type LocaleType = 'ru' | 'enUS' | 'de';

export type Corners = {
  [key in 'top-left' | 'bottom-left' | 'top-right' | 'bottom-right']?: boolean;
};

export type CalendarSize = 'big' | 'small';

export const SIZES = {
  big: 320,
  small: 248,
};

export const NAVIGATION_PANEL_SIZES = {
  big: 64,
  small: 52,
};

export const DAY_SIZES = {
  big: 40,
  small: 32,
};

export const DAY_PADDINGS = {
  big: 12,
  small: 8,
};

export const NAVIGATION_YEAR_SIZES = {
  big: 8,
  small: 6,
};

export const YEAR_SIZES = {
  big: 68,
  small: 56,
};

export const YEAR_PADDINGS = {
  big: 26,
  small: 20,
};

export const DAY_VIEW_PADDINGS = {
  big: '28px 20px 16px',
  small: '20px 12px 12px',
};

export const YEAR_VIEW_PADDINGS = {
  big: '28px 24px 16px',
  small: '20px 12px 12px',
};

export const HEADER_PADDINGS = {
  big: '26px 0px 28px 32px',
  small: '20px 0px 20px 20px',
};

export const PANEL_DAY_VIEW_PADDINGS = {
  big: '0 4px 28px 10px',
  small: '0 4px 20px 8px',
};

export const PANEL_YEAR_VIEW_PADDINGS = {
  big: '0 0 28px 6px',
  small: '0 4px 20px 8px',
};

export const DAY_NAMES_BOTTOM_MARGINS = {
  big: 12,
  small: 8,
};

export const DAY_NUMBERS = [0, 1, 2, 3, 4, 5, 6];

export const MENU_DAY = 'Выбор дня';
export const MENU_YEAR = 'Выбор года';

export const MENU_NEXT_YEAR = 'Вперед';
export const MENU_PREVIOUS_YEAR = 'Назад';

export const MENU_NEXT_MONTH = 'Следующий месяц';
export const MENU_PREVIOUS_MONTH = 'Предыдущий месяц';

export const DEFAULT_SIZE = 'big';
export const DEFAULT_YEAR_COUNT = 16;
export const DEFAULT_LOCALE_NAME = 'ru';
