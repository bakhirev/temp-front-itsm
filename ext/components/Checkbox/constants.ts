export type CheckboxSize = 'big' | 'small';

export const CHECKBOX_MIN_HEIGHT_SMALL = '20px';
export const CHECKBOX_MIN_HEIGHT_BIG = '24px';
export const MARK_BEFORE_OFFSET = '-9px';
export const MARK_HEIGHT_SMALL = '16px';
export const MARK_HEIGHT_BIG = '20px';
export const MARK_WIDTH_SMALL = '16px';
export const MARK_WIDTH_BIG = '20px';
export const CHECKBOX_MARGIN = '2px';
export const CHECKBOX_MARGIN_WITH_LABEL = '10px';
export const INDETERMINATE_SVG_WIDTH_SMALL = '8px';
export const INDETERMINATE_SVG_WIDTH_BIG = '10px';
export const BASIC_SVG_WIDTH_SMALL = '10px';
export const BASIC_SVG_WIDTH_BIG = '12px';
