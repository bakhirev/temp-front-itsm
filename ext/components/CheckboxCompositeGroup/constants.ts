export const ITEM_PADDING_LEFT_SMALL = '28px';
export const ITEM_PADDING_LEFT_BIG = '32px';
export const ITEM_MARGIN_BOTTOM_SMALL = '12px';
export const ITEM_MARGIN_BOTTOM_BIG = '16px';
export const ITEM_BEFORE_WIDTH_SMALL = '18px';
export const ITEM_BEFORE_WIDTH_BIG = '20px';
export const ITEM_BEFORE_LEFT_SMALL = '10px';
export const ITEM_BEFORE_LEFT_BIG = '12px';
export const COMPOSITEGROUP_PADDING_TOP_SMALL = '12px';
export const COMPOSITEGROUP_PADDING_TOP_BIG = '16px';
export const COMPOSITEGROUP_BEFORE_HEIGHT_SMALL = '12px';
export const COMPOSITEGROUP_BEFORE_HEIGHT_BIG = '14px';
export const COMPOSITEGROUP_BEFORE_LEFT_SMALL = '10px';
export const COMPOSITEGROUP_BEFORE_LEFT_BIG = '12px';
export const COMPOSITEGROUP_BEFORE_TOP = '2px';
