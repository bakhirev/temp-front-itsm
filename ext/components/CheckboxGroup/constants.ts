export const ITEM_MARGIN_BOTTOM_SMALL = '12px';
export const ITEM_MARGIN_BOTTOM_BIG = '16px';
