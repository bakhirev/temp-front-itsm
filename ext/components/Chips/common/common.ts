export type Size = 'big' | 'small';
