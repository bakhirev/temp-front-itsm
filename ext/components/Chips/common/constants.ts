export const ICON_SIZE_SMALL = 16;
export const ICON_SIZE_BIG = 20;
export const SHAPE_MICRO = '14px';
export const MAX_WIDTH = '400px';
export const HEIGHT_SMALL = '24px';
export const HEIGHT_BIG = '28px';
export const PADDING_BIG = '4px 0 4px 9px';
export const PADDING_SMALL = '4px 0 4px 8px';
export const TEXT_WIDTH_BIG = '358px';
export const TEXT_WIDTH_SMALL = '366px';
export const MARGIN_LEFT_BIG = '4px';
export const MARGIN_LEFT_SMALL = '2px';
export const IN_SELECT_MARGIN_SMALL = '8px';
export const IN_SELECT_MARGIN_BIG = '12px';
