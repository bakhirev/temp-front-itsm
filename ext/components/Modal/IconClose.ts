import styled from 'styled-components';

import { DEFAULT_THEME } from '../common';
import { MEDIA_MOBILE } from '../common/media';
import { markerStyle } from '../common/marker';

import {
  MOBILE_PADDING_TOP,
  MOBILE_PADDING_RIGHT,
  PADDING_TOP,
  SCROLL_PADDING_RIGHT,
  PADDING_RIGHT,
} from './constants';

export const IconClose = styled.div<{
  markerOffset: number;
  inverse?: boolean;
}>`
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: ${PADDING_TOP};
  right: calc(${PADDING_RIGHT} + ${SCROLL_PADDING_RIGHT});

  @media ${MEDIA_MOBILE} {
    top: ${MOBILE_PADDING_TOP};
    right: calc(${MOBILE_PADDING_RIGHT} + ${SCROLL_PADDING_RIGHT});
  }

  &:hover {
    cursor: pointer;
  }

  ${({ theme, markerOffset, inverse }) => `
    & svg {
      fill: ${inverse ? theme.color.neutral.white : theme.color.neutral[50]};
    }
    ${markerStyle(theme, markerOffset, inverse)}
  }`}
`;

IconClose.defaultProps = {
  theme: DEFAULT_THEME,
};
