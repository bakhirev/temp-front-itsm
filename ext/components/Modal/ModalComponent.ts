import styled from 'styled-components';

import { DEFAULT_THEME } from '../common';
import { MEDIA_MOBILE } from '../common/media';

import { Content } from './Content';
import {
  MOBILE_PADDING_LEFT,
  MOBILE_PADDING_TOP,
  PADDING_LEFT,
  PADDING_TOP,
  SCROLL_PADDING_RIGHT,
} from './constants';
import { calcContentHeight } from './utils';

interface IModalComponentProps {
  width: number;
  height: number;
  buttonCount: number;
}

export const ModalComponent = styled.div<IModalComponentProps>`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  outline: 0;
  padding: ${PADDING_TOP} ${SCROLL_PADDING_RIGHT} 0px ${PADDING_LEFT};

  @media ${MEDIA_MOBILE} {
    padding: ${MOBILE_PADDING_TOP} ${SCROLL_PADDING_RIGHT} 0px ${MOBILE_PADDING_LEFT};
  }

  ${({ theme, width, height, buttonCount }) => `
    width: ${width}px;
    max-height: ${height}vh;
    background-color: ${theme.color.neutral.white};
    box-shadow: ${theme.boxShadow[16]};
    border-radius: 8px;

    @media ${MEDIA_MOBILE} {
      width: 288px;
      max-height: 84vh;
    }

    ${Content} {
      max-height: ${calcContentHeight(height, buttonCount)};

      @media ${MEDIA_MOBILE} {
        max-height: ${calcContentHeight(height, buttonCount, true)};
      }
    }
  `}
`;

ModalComponent.defaultProps = {
  theme: DEFAULT_THEME,
};
