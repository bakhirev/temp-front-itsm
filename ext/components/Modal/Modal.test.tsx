import React, { ReactNode } from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import * as ContextModule from '../common/theme-context';

import { IconClose } from './IconClose';
import { ModalSize, SIZES_WIDTH, SIZES_HEIGHT } from './constants';
import { ModalComponent } from './ModalComponent';

import { Modal } from './';

const MODAL_HEADER = 'Header';
const MODAL_CONTENT = 'Content';

jest.mock('../common/default-theme', () => ({
  DEFAULT_THEME: {
    boxShadow: {
      16: '',
    },
    color: {
      neutral: {
        white: '',
      },
      opacity: {
        blackHover: '',
      },
    },
  },
}));

jest.spyOn(ContextModule, 'useThemeContext').mockImplementation(() => ({
  color: {
    opacity: {
      blackMedium: '#000000',
    },
  },
}));

jest.mock('../Typography', () => ({
  getTextStyle: () => '',
  // eslint-disable-next-line react/display-name
  H5: ({ children }: { children: ReactNode }) => <div>{children}</div>,
  // eslint-disable-next-line react/display-name
  Body1Long: ({ children }: { children: ReactNode }) => <div>{children}</div>,
}));

jest.mock('../Icons/service/CloseOutline.svg', () => ({
  // eslint-disable-next-line react/display-name
  ReactComponent: () => <div> x </div>,
}));

describe('Modal', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((overlay) => overlay && overlay.props.children) as any;
  });

  afterEach(() => {
    (ReactDOM.createPortal as any).mockClear();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render component non opened by default', () => {
    let wrapper = shallow(<Modal header={MODAL_HEADER}>{MODAL_CONTENT}</Modal>);
    expect(wrapper.find(ModalComponent).length).toBe(0);
    wrapper = shallow(
      <Modal opened header={MODAL_HEADER}>
        {MODAL_CONTENT}
      </Modal>
    );
    expect(wrapper.find(ModalComponent).length).toBe(1);
  });

  it('should render component with header and text content by default', () => {
    const wrapper = shallow(
      <Modal opened header={MODAL_HEADER}>
        {MODAL_CONTENT}
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with provided size', () => {
    const sizes: Array<ModalSize> = ['big', 'medium', 'small', 'micro'];
    sizes.forEach((size) => {
      const wrapper = shallow(
        <Modal opened header={MODAL_HEADER} size={size}>
          {MODAL_CONTENT}
        </Modal>
      );
      const modalComponent = wrapper.find(ModalComponent);
      expect(modalComponent.prop('width')).toEqual(SIZES_WIDTH[size]);
      expect(modalComponent.prop('height')).toEqual(SIZES_HEIGHT[size]);
    });
  });

  it('should render component with header and text content including primary button', () => {
    const wrapper = shallow(
      <Modal
        opened
        header={MODAL_HEADER}
        primaryButton={{
          children: 'Primary button',
          onClick: () => {
            /* do nothing */
          },
        }}
      >
        {MODAL_CONTENT}
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with header and text content including primary & secondary button', () => {
    const wrapper = shallow(
      <Modal
        opened
        header={MODAL_HEADER}
        primaryButton={{
          children: 'Primary button',
          onClick: () => {
            /* do nothing */
          },
        }}
        secondaryButton={{
          children: 'Secondary button',
          onClick: () => {
            /* do nothing */
          },
        }}
      >
        {MODAL_CONTENT}
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onRequestClose with stopPropagation when mouse has been clicked on "Close" icon', () => {
    let closeClicked = false;
    let stoppedPropagation = false;
    const clickMouseEventArgs = {
      stopPropagation: () => (stoppedPropagation = true),
    };
    const wrapper = shallow(
      <Modal opened header={MODAL_HEADER} onRequestClose={() => (closeClicked = true)}>
        {MODAL_CONTENT}
      </Modal>
    );
    const closeIcon = wrapper.find(IconClose);
    expect(stoppedPropagation).toBe(false);
    expect(closeClicked).toBe(false);
    closeIcon.simulate('click', clickMouseEventArgs);
    expect(stoppedPropagation).toBe(true);
    expect(closeClicked).toBe(true);
  });
});
