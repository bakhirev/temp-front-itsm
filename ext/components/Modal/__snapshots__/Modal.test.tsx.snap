// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render component with header and text content by default 1`] = `
<styled.div
  buttonCount={0}
  height={84}
  theme={
    Object {
      "boxShadow": Object {
        "16": "",
      },
      "color": Object {
        "neutral": Object {
          "white": "",
        },
        "opacity": Object {
          "blackHover": "",
        },
      },
    }
  }
  width={800}
>
  <styled.div>
    Header
  </styled.div>
  <styled.div
    markerOffset={6}
    onClick={[Function]}
    theme={
      Object {
        "boxShadow": Object {
          "16": "",
        },
        "color": Object {
          "neutral": Object {
            "white": "",
          },
          "opacity": Object {
            "blackHover": "",
          },
        },
      }
    }
  >
    <ReactComponent
      height={24}
      width={24}
    />
  </styled.div>
  <styled.div>
    Content
  </styled.div>
</styled.div>
`;

exports[`Modal should render component with header and text content including primary & secondary button 1`] = `
<styled.div
  buttonCount={2}
  height={84}
  theme={
    Object {
      "boxShadow": Object {
        "16": "",
      },
      "color": Object {
        "neutral": Object {
          "white": "",
        },
        "opacity": Object {
          "blackHover": "",
        },
      },
    }
  }
  width={800}
>
  <styled.div>
    Header
  </styled.div>
  <styled.div
    markerOffset={6}
    onClick={[Function]}
    theme={
      Object {
        "boxShadow": Object {
          "16": "",
        },
        "color": Object {
          "neutral": Object {
            "white": "",
          },
          "opacity": Object {
            "blackHover": "",
          },
        },
      }
    }
  >
    <ReactComponent
      height={24}
      width={24}
    />
  </styled.div>
  <styled.div>
    Content
  </styled.div>
  <styled.div>
    <Component
      kind="secondary"
      onClick={[Function]}
      size="small"
    >
      Secondary button
    </Component>
    <Component
      onClick={[Function]}
      size="small"
    >
      Primary button
    </Component>
  </styled.div>
</styled.div>
`;

exports[`Modal should render component with header and text content including primary button 1`] = `
<styled.div
  buttonCount={1}
  height={84}
  theme={
    Object {
      "boxShadow": Object {
        "16": "",
      },
      "color": Object {
        "neutral": Object {
          "white": "",
        },
        "opacity": Object {
          "blackHover": "",
        },
      },
    }
  }
  width={800}
>
  <styled.div>
    Header
  </styled.div>
  <styled.div
    markerOffset={6}
    onClick={[Function]}
    theme={
      Object {
        "boxShadow": Object {
          "16": "",
        },
        "color": Object {
          "neutral": Object {
            "white": "",
          },
          "opacity": Object {
            "blackHover": "",
          },
        },
      }
    }
  >
    <ReactComponent
      height={24}
      width={24}
    />
  </styled.div>
  <styled.div>
    Content
  </styled.div>
  <styled.div>
    <Component
      onClick={[Function]}
      size="small"
    >
      Primary button
    </Component>
  </styled.div>
</styled.div>
`;
